middlewares/authJwt.js

```diff
@@ -0,0 +1,40 @@
+const jwt = require("../services/jwt");
+const { resultMsg } = require("../constants");
+
+const authJWT = async (req, res, next) => {
+  const unauthorizedResult = {
+    result: resultMsg.fail,
+    message: resultMsg.unauthorized,
+  };
+
+  if (!req.headers.authorization) return res.status(401).json(unauthorizedResult);
+
+  const token = req.headers.authorization.split("Bearer ")[1];
+
+  try {
+    const myToken = await jwt.verify(token);
+
+    if (myToken) {
+      req.id = myToken.id;
+      req.email = myToken.email;
+      next();
+    } else {
+      return res.status(401).json(unauthorizedResult);
+    }
+  } catch (err) {
+    console.error(err);
+    if (err.message === "jwt expired") {
+      return res.status(500).json({
+        result: resultMsg.fail,
+        message: "토큰 유효기간이 만료되었습니다. 로그인이 필요합니다",
+      });
+    } else {
+      return res.status(500).json({
+        result: resultMsg.fail,
+        message: resultMsg.serverError,
+      });
+    }
+  }
```

Comment:
로직 정리가 조금 필요할 것 같습니다...! 제가 이해한 바로는 try-catch로 감싸고 try 안에서는 다시 if-else로 unauthorizedResult를 거르고 catch 에서는 같은 에러 코드를 메시지만 다르게 보내는 상황인 것 같습니다~! 아 그리고 제가 test할 때는 err.name === "TokenExpiredError"이면 만료된 토큰으로 나왔는데 message는 확인을 못해서 혹시 진경님께서 message를 확인하셨는지 한 번 여쭤봅니다🙇‍♀️

```suggestion
  try {
    //  try 구문으로 감싸져 있기 때문에 verify 되지 않는다면 바로 catch로 넘어갈 것 같습니다.
    const clientToken = await jwt.verify(token);
    req.id = clientToken.id;
    req.email = clientToken.email;
  
    next();
  } catch (err) {
    console.error(err);
    const message =
      err.name === "TokenExpiredError"
        ? resultMsg.tokenExpired
        : resultMsg.unauthorized;

    next(createError(401, message));
}
```

Reply:
넵 위에서 제안된 코드에 명시 하셨듯 name = "TokenExpiredError" 이고, 
message = "jwt expired" 입니다 (확인 완료 한건)

제안해주신 코드로 수정해서 커밋 올리겠습니다!